src/main/scala/net/snowflake/spark/snowflake/pushdowns/querygeneration/QueryBuilder.scala

```diff
@@ -131,6 +129,40 @@ private[querygeneration] class QueryBuilder(plan: LogicalPlan) {
     }
   }
 
+  def analyzeSparkPlan(plan: LogicalPlan): Unit = try {
+    plan match {
+      case _: ReturnAnswer =>
+        val statisticSet = mutable.HashSet[String]()
```

Comment:
Spark may attempt to pushdown sub-plan, so we only handle the plan with `ReturnAnswer` on the top.